packages/client/src/async-completion-client.ts

```diff
@@ -238,13 +247,17 @@ export class AsyncCompletionClient extends BaseClient {
         });
         cancelRequested = !!response.cancelRequested;
         paused = !!response.activityPaused;
+        reset = !!response.activityReset;
       }
     } catch (err) {
       this.handleError(err);
     }
     if (cancelRequested) {
       throw new ActivityCancelledError('cancelled');
     }
+    if (reset) {
```

Comment:
Should we be worried about this cascade of `if` statements? Can multiple of these conditions be true at the same time? If so, are we 100% sure about the order?

Reply:
Well, though more than one may be true, the way this is written where each branch throws, that's kind of irrelevant. I would make them elses for that reason I think.